### PR TITLE
Implement `JSON_ELEMENTS` IO format for responses

### PR DESCRIPTION
**Pull request by tailhook**
This is important for the non-interactive REPL mode, to allow fetching
large batches of the data without keeping everything in memory (and also
without replicating the whole JSONizing process on top of a binary
protocol).

I've tested this against patched Rust REPL. Need to implement similar feature in python bindings to have tests here.

### CODE REVIEW

File: docs/internals/protocol/messages.rst [hunk 1]
```diff
@@ -303,8 +303,16 @@ Format:
...
         BINARY = 0x62,
-        JSON = 0x6a
+
+        // Returns a single row and single field that contains
+        // a resultset as a single JSON array
+        JSON = 0x6a,
+
+        // Returns a single JSON string per row, preferred over JSON format
+        // because might be used for larger responses
+        JSON_ROWS = 0x4a,
```
Comment (elprans) [inline docs/internals/protocol/messages.rst]: Perhaps `JSON_ELEMENTS` is a better name?  We don't refer to result entries as "rows" generally.  (Sorry, I know I suggested the "rows" name earlier).

File: edb/pgsql/compiler/output.py [hunk 2]
```diff
@@ -525,5 +531,8 @@ def top_output_as_value(
 
         return stmt
 
+    elif env.output_format is context.OutputFormat.JSON_ROWS:
```
Comment (elprans) [inline edb/pgsql/compiler/output.py]: This branch seems redundant.

**Comment by tailhook** [comment]
Added a test

**Comment by tailhook** [comment]
@1st1, the test is here. I need your approval, as you have requested changes.

File: edb/server/compiler/compiler.py [hunk 3]
```diff
@@ -95,11 +95,23 @@ class CompileContext:
...
 
 
 pg_ql = lambda o: pg_common.quote_literal(str(o))
 
 
+def _convert_format(inp: enums.IoFormat) -> pg_compiler.OutputFormat:
+    try:
+        return _IO_FORMAT_MAP[inp]
+    except KeyError:
+        raise RuntimeError(f"unimplemented io format {inp!r}")
```
Comment (1st1) [inline edb/server/compiler/compiler.py]:
```suggestion
        raise RuntimeError(f"IO format {inp!r} is not supported")
```

**Pull request merged**